docs: Update Backfill Suggestions docs for opened/merged scope and 1,000-PR cap (#673)
Promptless/docs: src/content/docs/docs/configuring-promptless/triggers.mdx

Backfill suggestions lets you run Promptless on pull requests that already exist, so you can catch up on documentation for work that merged before a trigger was watching. It's available on saved pull request and merge request triggers: GitHub PRs, GitLab merge requests, and Bitbucket PRs.

> Open the **Backfill suggestions** panel on a trigger row, choose a window of the **last 14 days**, **last 30 days**, or **last 90 days**, and select **Preview**. The preview lists the pull requests that were opened or merged in that window and that the trigger matches, using the same repository, directory, and branch filters as the live webhook. Each result links to the pull request and shows whether it's open or merged. Backfill covers up to 1,000 pull requests per repository in the window.

Reviewer: ```suggestion
Open the **Backfill suggestions** panel on a trigger row, choose a window of the **last 14 days**, **last 30 days**, or **last 90 days**, and select **Preview**. The preview lists the pull requests that were opened or merged in that window and that the trigger matches, applying your repository, directory, and branch filters. Backfill covers up to 1,000 pull requests per repository in the window.
```

> Backfill matches pull requests that were opened or merged—it doesn't replay historical approvals. A trigger that fires only on first approval returns no backfill results, since approval matching applies only to live events going forward.

Reviewer: ```suggestion
Backfill matches pull requests that were opened or merged but it doesn't replay historical approvals. A trigger that fires only on first approval returns no backfill results because approval matching applies only to live events going forward.
```
